src/System.Memory/src/System/MemoryExtensions.Portable.cs

```diff
@@ -194,5 +195,119 @@ private static IntPtr MeasureStringAdjustment()
                 }
             }
         }
+
+        /// <summary>
+        /// Attempts to cast a ReadOnlySpan of one primitive type <typeparamref name="TFrom"/> to another primitive type <typeparamref name="TTo"/>.
+        /// </summary>
+        /// <param name="source">The source slice, of type <typeparamref name="TFrom"/>.</param>
+        /// <param name="output">The destination of type <typeparamref name="TTo"/>.</param>
+        /// <remarks>If <typeparamref name="TTo"/> is 8-byte aligned and <paramref name="source"/> points to a valid aligned <typeparamref name="TTo"/> address,
```

Comment:
For x86, it could be 4-byte aligned too, right? Update to word size/pointer size?

Reply:
Interesting point. Is the maximum alignment of a C# type on an x86 cpu 4 bytes? Seems like the word size wouldn't affect the alignment, but I'm not certain.

For some context, the shortcut here is that when a TFrom is aligned by 8-bytes then it is also aligned to 4-bytes, 2-bytes, 1-bytes. And since TTo can be at most aligned by 8-bytes, any TTo will also be aligned.